community/kernel/src/main/java/org/neo4j/procedure/UserFunction.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.procedure;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Declares a method as a Function, meaning the method can be called from the
+ * cypher query language.
+ * <p>
+ * Functions accept input, use that input to perform work, and then return a value. The work performed usually
+ * involves one or more resources, such as a {@link org.neo4j.graphdb.GraphDatabaseService}.
+ * <p>
+ * A function is associated with one of the following modes
+ *      READ    allows only reading the graph (default mode)
+ *      WRITE   allows reading and writing the graph
+ *      SCHEMA  allows reading the graphs and performing schema operations
+ *      DBMS    allows managing the database (i.e. change password)
+ *
```

Comment:
I thought we had said that functions are read only? Is the documentation out of sync?